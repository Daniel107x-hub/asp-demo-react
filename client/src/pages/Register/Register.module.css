form{
    margin: 1rem 0;
}

.register{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}