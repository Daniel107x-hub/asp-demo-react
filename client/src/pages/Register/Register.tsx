import React, { useEffect, useState } from 'react'
import styles from './Register.module.css'
import Card from '../../components/Card/Card'
import { toast } from 'react-toastify'
import { Link, Navigate, useNavigate } from 'react-router-dom'
import { useSelector } from 'react-redux'
import { RootState } from '../../app/store'
import { useRegisterMutation } from '../../services/User/UserService'

const Register = () => {
    const navigate = useNavigate();
    const [userData, setUserData] = useState({
        email: '',
        userName: '',
        password: ''
    });
    const [register, registerResult] = useRegisterMutation();

    const isAuthenticated = useSelector((state:RootState) => state.auth.isAuthenticated);

    const canSubmit = userData.email && userData.userName && userData.password;

    const handleInputChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
        const {name, value} = e.target;
        setUserData(state => {
            return {
                ...state,
                [name]: value
            }
        })
    }

    const handleRegister = (e: React.MouseEvent<HTMLButtonElement>) => {
        e.preventDefault();
        register(userData);
    }

    useEffect(() => {
        const {isError, isUninitialized} = registerResult;
        if(isUninitialized) return;
        if(isError){
            toast.error('Unable to register with the provided credentials :(');
            return;
        }
        toast.success('Sucessfully registered!');
        navigate('/login');
    }, [registerResult])

    if(isAuthenticated) return <Navigate to={'/todo'}/>
    return (
        <section className={styles.register}>
            <Card>
                <h1 className={styles.title}>Register</h1>
                <form>
                    <label htmlFor="email">Email: </label>
                    <input type="email" name="email" id="email" value={userData.email} onChange={handleInputChanged}/>
                    <label htmlFor="userName">Username: </label>
                    <input type="userName" name="userName" id="userName" value={userData.userName} onChange={handleInputChanged}/>
                    <label htmlFor="password">Password: </label>
                    <input type="password" name="password" id="password" value={userData.password} onChange={handleInputChanged} autoComplete='on password'/>
                    <button onClick={handleRegister} disabled={!canSubmit}>Submit</button>
                </form>
                <Link to={"/login"}>Already a user?</Link>
            </Card>
        </section>
    )
}

export default Register