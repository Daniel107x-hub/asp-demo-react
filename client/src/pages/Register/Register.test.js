import { renderWithProviders } from "../../utils/test-utils"
import Register from "./Register";

describe('Loads the resgistration screen', () => {
    it('Loads registration without breaking', () => {
        renderWithProviders(<Register/>);
    })

    it('Matches DOM snapshot', () => {
        const domeTree = renderWithProviders(<Register/>).asFragment();
        expect(domeTree).toMatchSnapshot();
    })
})