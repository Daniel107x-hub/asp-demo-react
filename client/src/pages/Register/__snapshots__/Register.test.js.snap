// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads the resgistration screen Matches DOM snapshot 1`] = `
<DocumentFragment>
  <section
    class="register"
  >
    <div
      class="card undefined"
    >
      <h1
        class="title"
      >
        Register
      </h1>
      <form>
        <label
          for="email"
        >
          Email: 
        </label>
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
        <label
          for="userName"
        >
          Username: 
        </label>
        <input
          id="userName"
          name="userName"
          type="userName"
          value=""
        />
        <label
          for="password"
        >
          Password: 
        </label>
        <input
          autocomplete="on password"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <button
          disabled=""
        >
          Submit
        </button>
      </form>
      <a
        href="/login"
      >
        Already a user?
      </a>
    </div>
  </section>
</DocumentFragment>
`;
